### ISSUE TITLE
Loss function for non-binary volumes

### ISSUE THREAD

**Issue by prameshgautam** [Project-MONAI/MONAI#1169]
I need to compare two images corresponding to landmark locations. I was thinking of something related to Dice loss. I cannot use dice loss since the image is not binary. The background is black but the landmark locations have varying intensities depending on the distance. What would be the best loss function to compare such images? Is cross-entropy suitable for this?

**Comment by Nic-Ma** [comment]
Hi @ericspod and @wyli ,

Could you guys help share some info here as you are experts on this topic?

Thanks.

**Comment by ericspod** [comment]
This might call for something more specific to landmarks. What I had done in the past with a regressor to predict landmark coordinates was use MAE loss between the ground truth and predicted positions, using some sort of cartesian distance loss didn't appear to help. If you have landmarks as single pixel/voxel values in an image converting these to coordinate values then using this kind of loss is what you might want to do. Dice certainly won't help here, I don't cross entropy would help since your images aren't really continuous fields.
